packages/hooks/src/useBackdropClick.ts

```diff
@@ -0,0 +1,78 @@
+import useEventListener from './useEventListener';
+
+/**
+ * Fires a callback when any element(s)
+ * _except_ those passed in as `foreground` is clicked.
+ *
+ * Note: Disable this hook (with the `enabled` arg)
+ * if the `foreground` element(s) are not in view (e.g. menu, tooltip, etc.).
+ */
+export function useBackdropClick(
+  /**
+   * Function called when any element other than those provided is clicked
+   */
+  callback: Function,
+
+  /**
+   * The primary element(s) that are excluded from backdrop click
+   */
+  foreground:
+    | React.RefObject<HTMLElement>
+    | Array<React.RefObject<HTMLElement>>,
+
+  /**
+   * Whether the callback is enabled.
+   * It's recommended to set this to `false` when not in use,
+   * and toggle to `true` when the main elements (menu, tooltip, etc) are visible
+   */
+  enabled = true,
+): void {
+  /**
+   * We add two event handlers to the document to handle the backdrop click behavior.
+   * Intended behavior is to close the menu, and keep focus on the component.
```

Comment:
Are these comments correct? This seems specific to a menu.

Reply:
Ah, yeah I ported this over from menu, Thought I had caught all of these